packages/core/src/DatadogProvider/Buffer/BufferSingleton.ts

```diff
@@ -0,0 +1,33 @@
+import { BoundedBuffer } from './BoundedBuffer';
+import type { DatadogBuffer } from './DatadogBuffer';
+import { PassThroughBuffer } from './PassThroughBuffer';
+
+export class BufferSingleton {
```

Comment:
Is this a common pattern in JS/TS to provide a static wrapper around a singleton to provide the singleton? I would normally see this as a "Provider" class.

Reply:
Yes, it's the most common way to implement the singleton pattern: https://refactoring.guru/design-patterns/singleton/typescript/example